yew/src/virtual_dom/vnode.rs

```diff
@@ -32,20 +33,38 @@ pub enum VNode {
 }
 
 impl VNode {
-    pub fn key(&self) -> &Option<String> {
+    pub fn key(&self) -> &Option<Key> {
         match self {
-            VNode::VTag(vtag) => &vtag.key,
-            VNode::VText(_) => &None,
             VNode::VComp(vcomp) => &vcomp.key,
             VNode::VList(vlist) => &vlist.key,
             VNode::VRef(_) => &None,
+            VNode::VTag(vtag) => &vtag.key,
+            VNode::VText(_) => &None,
+        }
+    }
+
+    pub fn reference(&self) -> Option<Node> {
+        match self {
+            VNode::VComp(vcomp) => vcomp.node_ref.get(),
+            VNode::VList(_) => None,
```

Comment:
We are going to have to rethink this since lists can be keyed themselves. Imagine you have a list of keyed lists that need to be re-ordered 😵

Reply:
I mitigated the issue, see the last commit (16ffa90). What do you think?